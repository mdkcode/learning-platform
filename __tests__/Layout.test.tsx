import { render } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import Layout from "@/app/dashboard/layout";
import { useSession } from "next-auth/react";
import { useRouter } from "next/navigation";

vi.mock("next-auth/react", () => ({
  useSession: vi.fn(),
}));

vi.mock("next/navigation", () => ({
  useRouter: vi.fn(),
}));

vi.mock("@/app/ui/dashboard/MobileHeader", () => ({
  MobileHeader: () => <div data-testid="mobile-header">MobileHeader</div>,
}));

vi.mock("@/app/ui/dashboard/sidenav", () => ({
  default: () => <div data-testid="side-nav">SideNav</div>,
}));

describe("Layout", () => {
  it("matches snapshot when session is valid", () => {
    (useSession as ReturnType<typeof vi.fn>).mockReturnValue({
      data: { user: { name: "Test User" }, expires: "future-date" },
    });

    const mockPush = vi.fn();
    (useRouter as ReturnType<typeof vi.fn>).mockReturnValue({ push: mockPush });

    const { container } = render(
      <Layout>
        <div>Test Child</div>
      </Layout>
    );

    expect(container).toMatchSnapshot();
  });
});
