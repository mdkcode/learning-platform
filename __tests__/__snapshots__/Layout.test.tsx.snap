// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout > matches snapshot when session is valid 1`] = `
<div>
  <div
    class="flex flex-col md:h-screen md:flex-row"
  >
    <div
      data-testid="mobile-header"
    >
      MobileHeader
    </div>
    <div
      data-testid="side-nav"
    >
      SideNav
    </div>
    <div
      class="flex-grow p-6 md:overflow-y-auto md:p-12"
    >
      <div>
        Test Child
      </div>
    </div>
  </div>
</div>
`;
